packages/expo-device/ios/DeviceModule.swift

```diff
@@ -0,0 +1,98 @@
+// swiftlint:disable:all number_separator
+
+import ExpoModulesCore
+import UIKit
+import MachO
+
+public class DeviceModule: Module {
+  public func definition() -> ModuleDefinition {
+    Name("ExpoDevice")
+
+    Constants([
+      "isDevice": isDevice(),
+      "brand": "Apple",
+      "manufacturer": "Apple",
+      "modelId": UIDevice.modelIdentifier,
+      "modelName": UIDevice.DeviceMap.modelName,
+      "deviceYearClass": UIDevice.DeviceMap.deviceYearClass,
+      "totalMemory": ProcessInfo.processInfo.physicalMemory,
+      "osName": UIDevice.current.systemName,
+      "osVersion": UIDevice.current.systemVersion,
+      "osBuildId": osBuildId(),
+      "osInternalBuildId": osBuildId(),
+      "deviceName": UIDevice.current.name,
+      "supportedCpuArchitectures": cpuArchitectures()
+    ])
+
+    AsyncFunction("getDeviceTypeAsync") { () -> Int in
+      switch UIDevice.current.userInterfaceIdiom {
+      case UIUserInterfaceIdiom.phone:
+        return DeviceType.phone.rawValue
+      case UIUserInterfaceIdiom.pad:
+        return DeviceType.tablet.rawValue
+      case UIUserInterfaceIdiom.tv:
+        return DeviceType.tv.rawValue
+      default:
+        return DeviceType.unknown.rawValue
+      }
+    }
+
+    AsyncFunction("getUptimeAsync") { () -> Double in
+      let uptimeMs: Double = ProcessInfo.processInfo.systemUptime * 1000
+
+      return uptimeMs
+    }
+
+    AsyncFunction("isRootedExperimentalAsync") { () -> Bool in
+      return UIDevice.current.isJailBroken
+    }
+  }
+}
+
+func isDevice() -> Bool {
+  #if targetEnvironment(simulator)
+    return false
+  #else
+    return true
+  #endif
+}
+
+func osBuildId() -> String? {
+  #if os(tvOS)
```

Comment:
Compiler directives in Swift usually don't increase the indent level (that's what Xcode does).
So I think the entire function needs to be re-indented.